src/hooks/useEncryptedStorage.js

```diff
@@ -29,6 +30,14 @@ export const useEncryptedStorage = () => {
     [getInstance]
   );
 
+  const setWallet = useCallback(
+    ({ phrase, priv, pub, addr, password }) => {
+      const instance = getInstance(password);
+      instance.setItem(WALLET, { phrase, priv, pub, addr });
```

Comment:
We'll want to modify this such that `phrase` and `priv` are stored in separate slots, since we'll want to retrieve them individually when needed. Also we should use a slightly different pattern, where `getStorage` returns an unlocked storage instance to the component, and then individual keys can be accessed as needed.  I'll put together an example off of this branch.